### ISSUE TITLE
[COMPILER REQUEST] MIR C

### ISSUE THREAD

**Issue by darko20** [compiler-explorer/compiler-explorer#3032]
**Compiler name:**
MIR C

**Compiler version:**
Latest release, currently 0.1.0

**Compiler language:**
C11, MIR intermediate representation

**Compiler homepage:**
https://github.com/vnmakarov/mir

**Motivation:**
This is a tiny, fast JIT based C compiler and IR targeting several platforms (see below) and is intended to be embedded and the source code generated on the fly. Being able to study the compiled output is particularly useful when the input source code is generated at runtime.

x86-64 Linux
x86-64 MacOS
aarch64 Linux
aarch64 (M1) MacOS
ppc64 big endian Linux
ppc64 little endian Linux
s390x Linux
RISCV Linux (I64MFD)

**Comment by mattgodbolt** [comment]
Hi! thanks for this: It looks like the MIR compiler generates only MIR (with the asm generated only at runtime when executing the program). Just checking that's what you were expecting to see: the MIR code?

**Comment by darko20** [comment]
That's correct, but you can get a compiled standalone program, for instance try the following in the package root dir:

```./c2m sieve.c && ./b2ctab <a.bmir >mir-ctab && gcc mir.o mir-gen.o -DMIR_USE_GEN mir-bin-driver.c -ldl```

But as you say native code is actually generated at run time so you'd have to capture the native code, which in ```mir-bin-driver.c``` is pointed to by ```fun_addr```.

I think seeing both the transformation of C to MIR code and MIR to assembly code would be optimal, but there doesn't seem to be an option to generate assembly code so in-memory disassembly would be required. Is that a showstopper?
